modular_skyrat/modules/medical/code/smartdarts.dm

```diff
@@ -0,0 +1,101 @@
+	to_chat(user, span_notice("You fill [src] with [trans] units of the solution. It now contains [reagents.total_volume] units."))
+
+	return SECONDARY_ATTACK_CONTINUE_CHAIN
+
+//The base smartdartgun
+/obj/item/gun/syringe/smartdart
+	name = "medical smartdart gun"
+	desc = "An adjusted version of the medical syringe gun that only allows smartdarts to be chambered."
+	w_class = WEIGHT_CLASS_NORMAL //I might need to look into changing this later depending on feedback
+	icon = 'modular_skyrat/modules/medical/icons/obj/dartguns.dmi'
+	icon_state = "smartdartgun"
+
+/obj/item/gun/syringe/smartdart/Initialize()
+	. = ..()
+	chambered = new /obj/item/ammo_casing/syringegun/dart(src)
+
+/obj/item/gun/syringe/smartdart/attackby(obj/item/container, mob/user, params, show_msg = TRUE)
+	if(istype(container, /obj/item/reagent_containers/syringe/smartdart))
+		..()
+	else
+		to_chat(user, span_notice("[container] is unable to fit inside of the [src]! Try using a smartdart instead."))
+		return FALSE
+
+//Smartdart projectiles
+/obj/item/ammo_casing/syringegun/dart
+	harmful = FALSE
+	projectile_type = /obj/projectile/bullet/dart/syringe/dart
+
+/obj/projectile/bullet/dart/syringe/dart
+	name = "smartdart"
+	damage = 0
+	var/list/allergyList = list()
+	var/prevention_used = FALSE
+	var/list/allowed_medicine = list(
+		/datum/reagent/medicine,
+		/datum/reagent/vaccine
+	)
+
+/obj/projectile/bullet/dart/syringe/dart/on_hit(atom/target, blocked = FALSE)
+	if(!iscarbon(target))
+		..(target, blocked)
+		reagents.flags &= ~(NO_REACT)
+		reagents.handle_reactions()
+		return BULLET_ACT_HIT
+	var/mob/living/carbon/injectee = target
+	if(!injectee.can_inject(target_zone = def_zone, injection_flags = inject_flags)) // if the syringe is blocked
+		blocked = 100
+	if(blocked == 100)
+		target.visible_message(span_danger("\The [src] is deflected!"), \
+							span_userdanger("You are protected against \the [src]!"))
+		return
+		//Checks to see if the target has allergies
+	for(var/datum/quirk/quirky as anything in injectee.quirks)
+		if(!istype(quirky, /datum/quirk/item_quirk/allergic))
+			continue
+		var/datum/quirk/item_quirk/allergic/allergies_quirk = quirky
+		allergyList = allergies_quirk.allergies
+			//The code that handles the actual injections
+	for(var/datum/reagent/meds in reagents.reagent_list)
+		if(!is_type_in_list(meds, allowed_medicine))
+			continue
+		if(is_type_in_list(meds, allergyList))
+			prevention_used = TRUE
+		else
+			injectee.reagents.add_reagent(meds.type, meds.volume)
+	injectee.visible_message(span_notice("[src] embeds itself into [injectee]"), span_notice("You feel a small prick as [src] embeds itself into you."))
+	if(prevention_used) //Used to signal that allergens were not injected into the target mob.
+		injectee.visible_message(span_notice("[src] lets out a short beep."), span_notice("You hear a short beep from [src]."))
+		playsound(loc, 'sound/machines/ping.ogg', 50, 1, -1)
+	return BULLET_ACT_HIT
+
```

Comment:
I'm reading this and I can't breathe.

Add some blank lines between your `if()` and `for()` statements, because otherwise this is just one big block of text and it's really hard to read.